src/Extension/MemberResetExtension.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace SilverStripe\MFA\Extension;
+
+use SilverStripe\Forms\FieldList;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\ORM\ValidationException;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\PasswordEncryptor_NotFoundException;
+use SilverStripe\Security\RandomGenerator;
+
+/**
+ * Provides DB columns / methods for account resets on Members
+ *
+ * @package SilverStripe\MFA\Extension
+ * @property Member owner
+ */
+class MemberResetExtension extends DataExtension
+{
+    private static $db = [
+        'AccountResetHash' => 'Varchar(160)',
+        'AccountResetExpired' => 'Datetime',
+    ];
+
+    public function updateCMSFields(FieldList $fields)
+    {
+        $fields->removeByName(['AccountResetHash', 'AccountResetExpired']);
+
+        return $fields;
+    }
+
+    /**
+     * Mirrors the implementation in Member::generateAutologinTokenAndStoreHash(),
+     * but against the AccountReset fields.
+     *
+     * @return string
+     * @throws PasswordEncryptor_NotFoundException
+     * @throws ValidationException
+     */
+    public function generateAccountResetTokenAndStoreHash(): string
+    {
+        $lifetime = $this->owner->config()->auto_login_token_lifetime;
+
+        do {
+            $generator = new RandomGenerator();
+            $token = $generator->randomToken();
+            $hash = $this->owner->encryptWithUserSettings($token);
+        } while (DataObject::get_one(Member::class, [
+            '"Member"."AccountResetHash"' => $hash,
+        ]));
+
+        $this->owner->AccountResetHash = $hash;
+        $this->owner->AccountResetExpired = date('Y-m-d H:i:s', time() + $lifetime);
```

Comment:
```suggestion
        $this->owner->AccountResetExpired = DBDatetime::create(DBDatetime::now() + $lifetime)-> Rfc2822();
```

It's a bit more long winded, but follows SS best practices. Related: https://github.com/silverstripe/silverstripe-framework/issues/8802